src/app/caesar-portal/group-list/group-list.component.ts

```diff
@@ -21,21 +22,45 @@ export class GroupListComponent implements OnInit {
   itemsPerPage = 10;
   currentPage = 1;
   groupsQuantity: number;
-
+  myGroupsFilter: boolean = false;
+  GroupProgressStatus : typeof GroupStatus = GroupStatus;
+  groupStatus : GroupStatus = GroupStatus.Current;
+ 
   constructor(private groupService: GroupService, private modalService: BsModalService) { }
 
   ngOnInit() {
-    this.groups = [];
+    this.showAllGroups();
+  }
 
+  private showAllGroups() {
     this.groupService.getAll().subscribe(
-      (data: Response) => {
-        this.groups = data.json();
+      (data: Group[]) => {
+        this.groups = data;
+        this.groupsQuantity = this.groups.length;
+        this.onPageChange(1);
+      },
+      error => console.log(error));
+  }
+
+  private showUserGroups() {
+    this.groupService.getUserGroups().subscribe(
+      (data: Group[]) => {
+        this.groups = data;
         this.groupsQuantity = this.groups.length;
         this.onPageChange(1);
       },
       error => console.log(error));
   }
 
+  public showMyGroups() {
+    this.myGroupsFilter ? this.showAllGroups() : this.showUserGroups();
```

Comment:
it is better to keep the value of relevant filter - first change the value and then render by it